Scanner stuff after a billion years

Arthur1asdf opened the pull request:
# Why

<!--
Please describe the motivation for this PR.
-->
Scanner features
# What
<!--
Please describe the changes made in this PR.
-->
Upon checkin hacker will get their class assigned to them using a least class assignment system. 
Added a pop up upon check in that will cover the entire screen + it's container that it is in, with their Full legal name, and added actions/instructions to take for the volunteers and organizers. 
Added discord roles for the hackers that check-in. 
Allow repeats for check in .
# Test Plan

<!--
Please describe how you tested the changes in this PR. 

This could include:
- How you verified the changes
- Screenshots or logs if applicable
- Any other relevant information
-->
<img width="771" height="850" alt="image" src="https://github.com/user-attachments/assets/c776d659-a52e-4c5d-8d7d-2c7f4a5f7ee5" />

using the await logs and console logs to check if my logic was correct

[video](https://www.youtube.com/watch?v=Tf2zSrWeDgU)
also if anyone with perms sees this could you remove the hacker roles off me? was not able to show getting discord hacker roles on myself, because I have the hacker roles.

TTRAN70 commented:
Please update test plan with videos/screenshots of frontend changes, thank you :)

Arthur1asdf commented:
Upon checkin hacker will get their class assigned to them using a least class assignment system.
If fastPass status is set to true before they check in or before the hackathon starts, they will get the vip class upon checking in to the check in event.
Added a pop up upon check in that will cover the entire container that it is in, with their Full legal name, and added actions/instructions to take for the volunteers and organizers along with their class. They will also get the KH8 role and their class role on discord.
Option for only specific classes to check-in to events
Allow repeats infinite for check in when toggled on .
VIP class allowed to check in as if they had all classes, but do not have repeat check in privilege unless toggled on like other hackers.  
I think that is it
New vid drop:
[video](https://youtu.be/RngJ8ryU3ds )

DVidal1205 reviewed (changes requested):
sorry for more but these are worth requesting

Arthur1asdf commented on apps/blade/src/app/admin/club/members/_components/scanner.tsx:
```diff
@@ -216,6 +292,20 @@ const ScannerPopUp = ({ eventType }: { eventType: "Member" | "Hacker" }) => {
+          <div className="fixed inset-0 z-[1000] flex w-full flex-col items-center justify-center gap-10 bg-black p-10 text-center text-3xl font-bold">
+            <div className="absolute top-10 w-56 text-lg">{checkInMessage}</div>
+            <div>{firstName}</div>
+            <div>{lastName}</div>
```
onSucess now gives eventName, which will be used to show the persistent dialog only on check in

Arthur1asdf commented:
[not vip](https://youtu.be/GAxF9s4EMLE)
[vip](https://youtu.be/pLjl_p2c4XU)

DVidal1205 reviewed (approved):
lgtm twin

Arthur1asdf commented:
Have not tested most recent changes with vip but everything should work
